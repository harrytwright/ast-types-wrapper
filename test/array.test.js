const chai = require('chai')
const recast = require('recast')
const { namedTypes: n, builders: b } = require('ast-types')

const assert = chai.assert

const { array, spread } = require('../src/array')
const { property } = require("../src/object");

// TODO: Maybe add more tests?? But these cover the basis
describe('object', function () {
  it('should create an array using an array', function () {
    const arr = array([1, 2, 3, 4])

    assert(n.ArrayExpression.check(arr), 'arr is an Array')
  });

  it('should create an array using an arrangement of elements', function () {
    const arr = array([1, { key: 'value' }, true, ['1', '2', 3, { key: { key: 'value' } }]])

    assert(n.ArrayExpression.check(arr), 'arr is an Array')
  });

  it('should create an array using an array with custom properties', function () {
    const arr = array.withProperties(
      spread('args')
    )

    assert(n.ArrayExpression.check(arr), 'arr is an Array')
  });
});
