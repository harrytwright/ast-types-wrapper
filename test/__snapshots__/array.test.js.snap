// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`object should create an array using an arrangement of elements 1`] = `
Object {
  "comments": null,
  "elements": Array [
    Object {
      "comments": null,
      "loc": null,
      "regex": null,
      "type": "Literal",
      "value": 1,
    },
    Object {
      "comments": null,
      "loc": null,
      "properties": Array [
        Object {
          "comments": null,
          "computed": false,
          "decorators": null,
          "key": Object {
            "comments": null,
            "loc": null,
            "name": "key",
            "optional": false,
            "type": "Identifier",
            "typeAnnotation": null,
          },
          "kind": "init",
          "loc": null,
          "method": false,
          "shorthand": false,
          "type": "Property",
          "value": Object {
            "comments": null,
            "loc": null,
            "regex": null,
            "type": "Literal",
            "value": "value",
          },
        },
      ],
      "type": "ObjectExpression",
    },
    Object {
      "comments": null,
      "loc": null,
      "regex": null,
      "type": "Literal",
      "value": true,
    },
    Object {
      "comments": null,
      "elements": Array [
        Object {
          "comments": null,
          "loc": null,
          "regex": null,
          "type": "Literal",
          "value": "1",
        },
        Object {
          "comments": null,
          "loc": null,
          "regex": null,
          "type": "Literal",
          "value": "2",
        },
        Object {
          "comments": null,
          "loc": null,
          "regex": null,
          "type": "Literal",
          "value": 3,
        },
        Object {
          "comments": null,
          "loc": null,
          "properties": Array [
            Object {
              "comments": null,
              "computed": false,
              "decorators": null,
              "key": Object {
                "comments": null,
                "loc": null,
                "name": "key",
                "optional": false,
                "type": "Identifier",
                "typeAnnotation": null,
              },
              "kind": "init",
              "loc": null,
              "method": false,
              "shorthand": false,
              "type": "Property",
              "value": Object {
                "comments": null,
                "loc": null,
                "properties": Array [
                  Object {
                    "comments": null,
                    "computed": false,
                    "decorators": null,
                    "key": Object {
                      "comments": null,
                      "loc": null,
                      "name": "key",
                      "optional": false,
                      "type": "Identifier",
                      "typeAnnotation": null,
                    },
                    "kind": "init",
                    "loc": null,
                    "method": false,
                    "shorthand": false,
                    "type": "Property",
                    "value": Object {
                      "comments": null,
                      "loc": null,
                      "regex": null,
                      "type": "Literal",
                      "value": "value",
                    },
                  },
                ],
                "type": "ObjectExpression",
              },
            },
          ],
          "type": "ObjectExpression",
        },
      ],
      "loc": null,
      "type": "ArrayExpression",
    },
  ],
  "loc": null,
  "type": "ArrayExpression",
}
`;

exports[`object should create an array using an array 1`] = `
Object {
  "comments": null,
  "elements": Array [
    Object {
      "comments": null,
      "loc": null,
      "regex": null,
      "type": "Literal",
      "value": 1,
    },
    Object {
      "comments": null,
      "loc": null,
      "regex": null,
      "type": "Literal",
      "value": 2,
    },
    Object {
      "comments": null,
      "loc": null,
      "regex": null,
      "type": "Literal",
      "value": 3,
    },
    Object {
      "comments": null,
      "loc": null,
      "regex": null,
      "type": "Literal",
      "value": 4,
    },
  ],
  "loc": null,
  "type": "ArrayExpression",
}
`;

exports[`object should create an array using an array with custom properties 1`] = `
Object {
  "comments": null,
  "elements": Array [
    Object {
      "argument": Object {
        "comments": null,
        "loc": null,
        "name": "args",
        "optional": false,
        "type": "Identifier",
        "typeAnnotation": null,
      },
      "comments": null,
      "loc": null,
      "type": "SpreadElement",
    },
  ],
  "loc": null,
  "type": "ArrayExpression",
}
`;
